store/postgres/secret_repository.go

```diff
@@ -84,13 +106,24 @@ func (repo *secretRepository) Insert(ctx context.Context, resource models.Projec
 }
 
 func (repo *secretRepository) Save(ctx context.Context, spec models.ProjectSecretItem) error {
-	existingResource, err := repo.GetByName(ctx, spec.Name)
+	_, err := repo.GetByName(ctx, spec.Name)
 	if errors.Is(err, store.ErrResourceNotFound) {
 		return repo.Insert(ctx, spec)
 	} else if err != nil {
 		return errors.Wrap(err, "unable to find secret by name")
 	}
-	resource, err := Secret{}.FromSpec(spec, repo.project, repo.hash)
+	return errors.New("secret already exist")
```

Comment:
We should generalize this error to `resource already exists` or something similar to `store.ErrResourceNotFound` or should get rid of this logic of checking if it exists and only insert the secret? Can you check if gorm throws an error if it already exists?
Because currently, I think we are leaking our implementation detail out of an interface.

Reply:
Gorm will throw an error like 
```
\x1b[31;1m /../optimus/store/postgres/secret_repository.go:135 \x1b[35;1mERROR: duplicate key value violates unique constraint \"secret_project_id_name_key\" (SQLSTATE 23505)" ....(prints whole query with values) 
```

Which will return the implementation (and query) details back in the api, so we will need to replace the returned error.

Can you please explain how are we leaking implementation details out of interface ?

Reply:
checking explicitly is fine & we don't need to rely on gorm error.